### PR TITLE
CB-2567 arm template for db stack creation

### PR DESCRIPTION
**Pull request by gergopapi2**
This is the single server deployment of postgres (is HA).
Template for server group is still to be checked if needed, but will come later. 

I provide some readings for the template I created:

arm template for postgres:
https://docs.microsoft.com/en-us/azure/templates/microsoft.dbforpostgresql/2017-12-01/servers

service endpoints: vnet to postgres connection without adding the postgres a public ip:
https://docs.microsoft.com/en-us/azure/postgresql/howto-manage-vnet-using-portal

### CODE REVIEW

File: cloud-azure/src/main/resources/templates/arm-dbstack.ftl [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                "sslEnforcement": "[parameters('sslEnforcement')]",
+                "storageProfile": {
+                    "backupRetentionDays": "[parameters('backupRetentionDays')]",
+                    "geoRedundantBackup": "[parameters('geoREdundantLookup')]",
+                    "storageMB": "[parameters('skuSizeMB')]",
+                    "storageAutogrow": "Disabled"
+                },
+                "createMode": "Default"
+            },
+            "location": "NorthEurope",
```
Comment (bhavanki) [inline cloud-azure/src/main/resources/templates/arm-dbstack.ftl]: Should be "[parameters('location')]" ?

File: cloud-azure/src/main/resources/templates/arm-dbstack.ftl [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            "type": "securestring",
+            "minLength": 8,
+            "maxLength": 128,
+            "metadata": {
+                "description": "The administrator password for the database server."
+            }
+        },
+        "sslEnforcement": {
+            "type": "string",
+            "defaultValue": "Enabled",
```
Comment (bhavanki) [inline cloud-azure/src/main/resources/templates/arm-dbstack.ftl]: What do you think about making this a Boolean value? Is it possible and straightforward to have the template engine convert true/false to Enabled/Disabled? I worry that the spelling will be wrong at some point. :)

Same for geoRedundantLookup, storageAutogrow.

File: cloud-azure/src/main/resources/templates/arm-dbstack.ftl [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    "$schema": "http://schema.management.azure.com/schemas/2014-04-01-preview/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "dbServerName": {
+            "type": "string",
+            "metadata": {
+                "description": "Name of the database server resource."
+            }
+        },
+        "skuTier": {
```
Comment (jadair-cloudera) [inline cloud-azure/src/main/resources/templates/arm-dbstack.ftl]: Add description metadata?

File: cloud-azure/src/main/resources/templates/arm-dbstack.ftl [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+                "GP_Gen5_16",
+                "GP_Gen5_32",
+                "MO_Gen5_2",
+                "MO_Gen5_4",
+                "MO_Gen5_8",
+                "MO_Gen5_16",
+                "MO_Gen5_32"
+            ],
+            "metadata": {
+                "description": "Azure database for PostgreSQL sku name "
```
Comment (jadair-cloudera) [inline cloud-azure/src/main/resources/templates/arm-dbstack.ftl]: NIT: SKU

File: cloud-azure/src/main/resources/templates/arm-dbstack.ftl (hunk 1 above)
Comment (gergopapi2) [reply cloud-azure/src/main/resources/templates/arm-dbstack.ftl]: Thanks, corrected.

File: cloud-azure/src/main/resources/templates/arm-dbstack.ftl (hunk 2 above)
Comment (gergopapi2) [reply cloud-azure/src/main/resources/templates/arm-dbstack.ftl]: Yes, would be nice, I check it now. On the other hand, field "allowedValues" does a sanity check on the input string right away. The feedback comes before anything in azure would be provisioned.

Comment (gergopapi2) [reply cloud-azure/src/main/resources/templates/arm-dbstack.ftl]: Ok, it was quite easy to do the boolean mapping, so fixed.

File: cloud-azure/src/main/resources/templates/arm-dbstack.ftl (hunk 3 above)
Comment (gergopapi2) [reply cloud-azure/src/main/resources/templates/arm-dbstack.ftl]: Yes, a bunch of descriptions were missing, thanks for pointing out that.

**Pull request merged**